Ball of Duty Server/Domain/Map.cs

```diff
@@ -169,5 +172,32 @@ public void UpdatePosition(Point position, int goId)
                 go.Body.Position = position;
             }
         }
+        /*
+
+        */
+        public void Update(Observable observable)
+        {
+            throw new NotImplementedException();
+        }
+        /*
+        Called when an observable object calls the overloaded NotifyObservers which takes data. 
+        If observable.hp<1, killer is found using data (which should be killerID) 
+        On the killer character AddKill is then called
+        */
+        public void Update(Observable observable, object data)
+        {
+            
+            Character victim = observable as Character;
+            if (victim.HP < 1)
+            {
+                int killerID = Convert.ToInt32(data);
+                GameObject killer;
+                bool succeded = GameObjects.TryGetValue(killerID, out killer);
+                Character killerFinal = killer as Character;
```

Comment:
Would if (succedd) {code using killer} else {fail message} be correct?
(I'm guessing you mean the ID might not be correct and we then wouldn't get a GameObject)